docs/topics/multiplatform/multiplatform-discover-project.md

```diff
@@ -1,169 +1,315 @@
+This way, each multiplatform project defines a set of supported targets. See
+the [Hierarchical project structure](multiplatform-hierarchy.md) section to learn more about declaring targets in your
+build scripts.
+
+With the `jvm` and `iosArm64` targets declared, the common code in `commonMain` will be compiled to these targets:
+
+![Targets](target-diagram.svg){width=700}
+
+To understand which code is going to be compiled to a specific target, you can think of a target as a label attached
+to Kotlin source files. Kotlin uses these labels to determine how to compile your code, which binaries to produce, and
+which language constructions and dependencies are allowed in that code.
+
+If you want to compile the `greeting.kt` file to `.js` as well, you only need to declare the JS target. The code
+in `commonMain` then receives an additional `js` label, corresponding to the JS target, which instructs Kotlin to
+produce `.js` files:
+
+![Target labels](target-labels-diagram.svg){width=700}
+
+That's how the Kotlin compiler works with the common code compiled to all the declared targets.
+See [Source sets](#source-sets) to learn how to write platform-specific code.
 
 ## Source sets
 
-The project includes the directory `src` with Kotlin source sets, which are collections of Kotlin code files, along with 
-their resources, dependencies, and language settings. A source set can be used in Kotlin compilations for one or more 
-target platforms. 
+A _Kotlin source set_ is a set of source files with its own targets, dependencies, and compiler options. It's the main
+way to share code in multiplatform projects.
 
-Each source set directory includes Kotlin code files (the `kotlin` directory) and `resources`. The project wizard creates 
-default source sets for the `main` and `test` compilations of the common code and all added targets. 
+Each source set in a multiplatform project:
 
-![Source sets](source-sets.png){width=300}
+* Has a name that is unique for a given project.
+* Contains a set of source files and resources, usually stored in the directory with the name of the source set.
+* Specifies a set of targets to which the code in this source set compiles. These targets impact which language
+  constructions and dependencies are available in this source set.
+* Defines its own dependencies and the compiler options.
 
-> Source set names are case-sensitive.
->
-{type="note"}
+Kotlin provides a bunch of predefined source sets. One of them is `commonMain`, which is present in all multiplatform
+projects and compiles to all declared targets.
+
+You interact with source sets as directories inside `src` in Kotlin Multiplatform projects:
 
-Source sets are added to the `sourceSets` block of the top-level `kotlin` block. For example, this is the source sets
-structure you get when creating a multiplatform library with the IntelliJ IDEA project wizard:
+![Shared sources](src-directory-diagram.png){width=350}
```

Comment:
Would be useful to explain, that the picture shows `commonMain`, `iosMain` and `jvmMain` source sets